File: Corte1/RetosEntrega19Agosto/RetoDiagonal.cpp
#include <iostream>
using namespace std;

int main() {
    int tamanoMatriz = 0;
    int sumaDiagonalPrincipal = 0;
    int sumaDiagonalSecundaria = 0;
    int valorMatriz = 0;
    cout << "Ingrese el tamaño de la matriz: ";
    cin >> tamanoMatriz;

    int matriz[tamanoMatriz][tamanoMatriz];

    for (int i = 0; i < tamanoMatriz; i++) {
        for (int j = 0; j < tamanoMatriz; j++) {
            cout << "Ingrese el valor para la posición [" << i << "][" << j << "]: ";
            cin >> valorMatriz;
            matriz[i][j] = valorMatriz;
        }
    }

    for (int i = 0; i < tamanoMatriz; i++) {
        sumaDiagonalPrincipal += matriz[i][i];
        sumaDiagonalSecundaria += matriz[i][tamanoMatriz - 1 - i];
    }
    if (sumaDiagonalPrincipal == sumaDiagonalSecundaria) {
        cout << "La suma de la diagonal principal es igual a la suma de la diagonal secundaria." << endl;
    } else {
        cout << "La suma de la diagonal principal no es igual a la suma de la diagonal secundaria." << endl;
    }
    cout << "Suma de la diagonal principal: " << sumaDiagonalPrincipal << endl;
    cout << "Suma de la diagonal secundaria: " << sumaDiagonalSecundaria << endl;
    return 0;
}

File: Corte1/RetosEntrega19Agosto/RetoDiagonal.py
# Se inicializan variables
tamanoMatriz = 0
tamanoMatriz = int(input("Ingrese el tamaño de la matriz: "))
valorMatriz = 0
sumaDiagonalPrincipal = 0
sumaDiagonalSecundaria = 0

matriz = []

#El usuario rellena la matriz con los valores que quiera, se usa .append para agregar los valores a la matriz, ya que en otro caso da error debido a que esta vacia
for i in range(tamanoMatriz):
    fila = []
    for j in range(tamanoMatriz):
        valorMatriz = int(input(f"Ingrese el valor de la posición [{i}][{j}]: "))
        fila.append(valorMatriz)
    matriz.append(fila)

# Se recorre la matriz sumando los valores de la diagonal principal y secundaria
#Se optimiza el recorrido usando un solo for, ya que ambas se pueden con i, solo toca hacer la operacion para la secundaria
for i in range(tamanoMatriz):
    sumaDiagonalPrincipal += matriz[i][i]
    sumaDiagonalSecundaria += matriz[i][tamanoMatriz - 1 - i]

if sumaDiagonalPrincipal == sumaDiagonalSecundaria:
    print("La diagonal principal y secundaria son iguales.")
else:
    print("La diagonal principal y secundaria no son iguales.")
print(f"La suma de la diagonal principal es: {sumaDiagonalPrincipal}")
print(f"La suma de la diagonal secundaria es: {sumaDiagonalSecundaria}")
